### DISCUSSION TITLE
Enter Pem pass phrase

### DISCUSSION THREAD

**Discussion by InfiniteLife** [cesanta/mongoose#2052] [Ask questions, get answers] [answered]
I'm trying to enable https for web server a in tutorial:
```
static void fn(struct mg_connection *c, int ev, void *ev_data, void *fn_data) {
  if (ev == MG_EV_ACCEPT) {
    struct mg_tls_opts opts = {
      .cert = "cert.pem",    // Certificate file
      .certkey = "key.pem",  // Private key file
    };
    mg_tls_init(c, &opts);
  }
```

But when I run the server and try to access it from browser, console running server application promts "Enter PEM pass phrase" and waits for promt. How do I pass file that contains pass phrase to web server, so it wont ask for pem phrase?

**Comment by scaprile** [comment, marked as answer]
There is no API for that, use an unlocked certificate as the examples show

**Reply by InfiniteLife** [reply]
Got it. My company uses password to encrypt private key. I was able to decrypt it, but Im wondering why it is not a part of API? Private key encryption isnt used widely or some other reason?

**Reply by scaprile** [reply]
AFAIK most use is in embedded systems so there is no keyboard nor user. Keys are generated for each device and either storage (flash) is encrypted/protected or the compromised keys are revoked and new ones generated.

I suggest you get in touch with our business people and discuss your usage case: support@cesanta.com
